src/app/image-group-page/image-group-page.component.ts

```diff
@@ -428,11 +438,41 @@ export class ImageGroupPage implements OnInit, OnDestroy {
       })
       .catch( error => {
         console.error(error)
-        this.exportLoadingStateopts.state = LoadingState.error
-        this.exportLoadingStateopts.errorType = 'server'
+        clearInterval(this.loadingStateInterval)
+        this.pollForExportStatus()
       })
   }
 
+  private pollForExportStatus(): void {
+    // When the server timesout or there is an error begin polling on export status every 15s
+    this.exportStatusInterval = setInterval( () => {
+      this.checkExportStatus(this.ig.id)
+    }, 15000)
+  }
+
+  private checkExportStatus(groupId: string): void {
+    this._ig.checkExportStatus(groupId).subscribe(
+      (response) => {
+        console.log('The response is: ', response)
+        if(response.status && response.status === 'COMPLETED') {
+          clearInterval(this.exportStatusInterval)
+          clearInterval(this.loadingStateInterval)
+          let downloadLink = this._auth.getThumbHostname() + response.path.replace('/nas/', '/thumb/')
+
+          this.exportLoadingStateopts.progress = 100
+          this.exportLoadingStateopts.state = LoadingState.completed
+          // On success fade out the component after 5 sec & begin download
+          setTimeout(() => {
+            this.closeExportLoadingState()
+            this.downLoadFile(this.ig.name, downloadLink)
+          }, 5000)
+        }
+      }, (error) => {
+        console.error('Error in check status: ', error)
+      }
+    )
+  }
+
   public closeExportLoadingState(event?: any): void {
```

Comment:
@rehanabbasi I'm not sure of the best approach to take here. But, canceling the export now needs logic to clear the series of calls to the check endpoint.
```
clearInterval(this.exportStatusInterval)
clearInterval(this.loadingStateInterval)
```